src/mlops/orchestration/orchestration_caller.py

```diff
@@ -0,0 +1,363 @@
+    # validate orchestrator string
+    if orc is None:
+        logging.info("Orchestrator name not passed")
+        return False
+    if orc.lower() not in APPROVED_ORCHESTRATORS:
+        logging.info(f"{orc} is not an approved orchestrator for the pipeline")
+        return False
+
+    # validate orchestrator endpoint
+    if orc_endpoint is None:
+        logging.info("Orchestrator endpoint not passed")
+        return False
+    # try to ping orchestrator to make sure it is up
+    command = ["ping", "-c", "1", orc.lower()]
+    if subprocess.call(command) == 0:
+        return True
+    command = ["ping", "-c", "1", orc_endpoint]
+    if subprocess.call(command) == 0:
+        return True
+    logging.info(f"Unable to ping orchestrator at {orc_endpoint}")
+    return False
+
+
+# ------------------------------------------------
+# Message Classes
+# ------------------------------------------------
+
+
+class Producer:
+    def __init__(
+        self,
+        topic: str or List[str] = None,
+        orchestrator: str = None,
+        orchestrator_endpoint: str = None,
+        **kwargs,
+    ):
+        """
+        Initialize the higher level Producer class exposed to the user
+
+        :param topic: the event bucket this producer writes to. Should not
+                contain any white spaces. These will be removed.
+        :type topic: str or List[str]
+        :param orchestrator: the name of the orchestrator in use
+        :type orchestrator: str
+        :param orchestrator_endpoint: The endpoint of the orchestrator
+        :type orchestrator_endpoint: str
+        :param `**kwargs`: any additional configuration requirements desired
+                for the Producer. Keys based on underlying orchestration
+                requirements. Look at the docs for the orchestrator wrapper
+                to determine desired requirements
+        :type `**kwargs`: Any
+        """
+        # determine underlying orchestrator
+        if _test_passed_orchestrator(orchestrator_endpoint, orchestrator):
+            self.orchestrator = orchestrator.lower()
+            self.orchestrator_endpoint = orchestrator_endpoint
+        else:
+            (
+                self.orchestrator,
+                self.orchestrator_endpoint,
+            ) = _get_orchestrator()
+
+        # connect to orchestrator and create the producer
+        if self.orchestrator == "kafka":
+            if topic is not None:
+                self.topic = _kafka_format_topic(topic)
+            else:
+                self.topic = KAFKA_TOPIC
+            self.producer = _kafka_producer_connect(
+                self.orchestrator_endpoint, **kwargs
+            )
+        else:
+            logging.warning(
+                "No orchestrator. Pipeline tools will have to be run manually."
+            )
+            self.producer = None
+            self.topic = None
+
+    def send(
+        self, message: MLOPS_Message = MLOPS_Message(), flush: bool = False

```

Comment:
Is there a way to implement the top-level `Producer` and `Consumer` types without coupling it with the messaging interface?